zap/src/main/dist/scripts/templates/sitemodifier/Site Modifier default template.js

```diff
@@ -0,0 +1,68 @@
+/*
+Site modifier scripts allow you to change how requests are represented in the sites tree.
+The sites tree is ZAP's internal representation of the applications. The closer this matches the applications's functionality the better.
+*/
+
+var ArrayList = Java.type('java.util.ArrayList')
+var DefaultNameValuePair = Java.type('org.zaproxy.zap.model.DefaultNameValuePair')
+
+/**
+ * Gets the parameters of the given {@code type} from the given {@code message}. Returning null
+ * is taken to mean use the default methods for obtaining parameters.
+ * The parameter names are shown in brackets after the site tree node names.
+ *
+ * The parameters are split using the key value pair separator(s) and each resulting
+ * parameter is split into name/value pairs using key value separator(s).
+ *
+ * Parameters' names and values are in decoded form.
+ *
+ * @param msg the message whose parameters will be extracted from. This is an HttpMessage object.
+ * @param type the type of parameters to extract. This is an HtmlParameter.Type enum which can be one of cookie, form, url, header, multipart
+ * @param helper a helper class, currently unused
+ * @return a {@code List} containing the parameters as DefaultNameValuePair objects
+ * @throws IllegalArgumentException if the {@code msg} or {@code type} is {@code null}.
+ */
+
+function getParameters(msg, type, helper) {
+	print("get params " + msg.getRequestHeader().getURI() + " " + type)
+	var list = new ArrayList()
+	switch (type.name()) {
+		case 'url': 
+			break
+		case 'form': 
+			break
+		default: 
+			break
+	}
```

Comment:
Just default or remove since all break?

Reply:
I'd rather leave them in - it shows what implementors typically need to do...

Reply:
Add some comment/explanation then?